exchange/utils_test.go

```diff
@@ -1,11 +1,137 @@
 package exchange
 
 import (
+	"context"
+	"encoding/json"
 	"testing"
 
+	"github.com/mxmCherry/openrtb"
 	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/pbsmetrics"
+	"github.com/stretchr/testify/assert"
 )
 
+// permissionsMock mocks the Permissions interface for tests
+//
+// It only allows appnexus for GDPR consent
+type permissionsMock struct{}
```

Comment:
A very similar class exists in other tests: https://github.com/prebid/prebid-server/blob/master/endpoints/cookie_sync_test.go#L187

This is one of those things I really don't like about Go... various language rules conspire to make it really hard to write shareable test code that's clearly _test_ code, and not meant for prod use.

The stdlib pattern is to make a new package. So for example mock stuff for `http` is stored in `http/httptest`: https://golang.org/pkg/net/http/httptest/

Might be worth doing that with this too, to avoid repeated code.

Reply:
Ha! Thanks for pointing that out. Didn't know there existed another mock for that interface. And now that I am looking more closely, it seems there are a couple more mocks for that interface.

https://github.com/prebid/prebid-server/blob/master/endpoints/auction_test.go#L398
https://github.com/prebid/prebid-server/blob/master/endpoints/setuid_test.go#L178

I will try to combine these mocks into one. Also, I'd like to know thoughts about using [testify's mock](https://godoc.org/github.com/stretchr/testify/mock). I think it has some nice mocking features that we can leverage.

Reply:
I tried combining all those different mocks of the `Permissions` interface in one but I think that will require quite a bit of refactoring and shouldn't be part of this PR so will leave it out of this one.

Reply:
I had no idea that `mock` package existed... it looks as well designed as the rest of the `testify` project is.

Definitely could be useful. I can't really judge how useful without trying it in a few places.